types/lazy/lazy.go

```diff
@@ -4,7 +4,17 @@
 // Package lazy provides types for lazily initialized values.
 package lazy
 
-import "sync"
+import (
+	"reflect"
+	"sync"
+	"sync/atomic"
+
+	"tailscale.com/types/ptr"
+)
+
+// nilErrPtr is a sentinel *error value for SyncValue.err to signal
+// that SyncValue.v is valid.
+var nilErrPtr = reflect.New(reflect.TypeFor[error]()).Interface().(*error)
```

Comment:
OOC, why use `reflect.New` here?

Reply:
It's the first thing I thought of. What were you thinking of?

Reply:
Oh, `var nilErrPtr = ptr.To((error)(nil))` works too

Reply:
Yeah that 😃

Reply:
or `var nilErrPtr = ptr.To[error](nil)` is a bit prettier